File: src/wms/model/client/type/Supplier.java
package wms.model.client.type;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.Table;

import wms.model.client.Client;

@Entity
@Table(name = "client")
@DiscriminatorValue(value = "1")
public class Supplier extends Client {
	private static final long serialVersionUID = -3487771558214472341L;

	public Supplier() {
		super();
	}

}

File: application/src/main/webapp/resources/app/model/entity/Supplier.js
/**
 * Created with JetBrains WebStorm.
 * User: kornicameister
 * Date: 04.11.12
 * Time: 01:25
 * To change this template use File | Settings | File Templates.
 */

Ext.define('WMS.model.entity.Supplier', {
    extend: 'WMS.model.entity.Client',
    proxy : {
        type       : 'wms',
        url        : 'wms/agent/client',
        extraParams: {
            type: 'supplier'
        },
        writer     : {
            type          : 'json',
            root          : 'data',
            allowSingle   : false,
            writeAllFields: false,
            getRecordData : function (record, operation) {
                return WMS.model.entity.Client.getRecordData(record, operation);
            }
        }
    }
});